File: src/fizzbuzz.pl
# 1から100まで繰り返す --- (*1)
for my $i (1..100) {
  print fizzbuzz($i)."\n";
}

# fizzbuzzに応じた値を返す --- (*2)
sub fizzbuzz {
  my $i = $_[0];
  my $is_fizz = ($i % 3 == 0);
  my $is_buzz = ($i % 5 == 0);
  return 'FizzBuzz' if ($is_fizz && $is_buzz);
  return 'Fizz' if ($is_fizz);
  return 'Buzz' if ($is_buzz);
  $i; # --- (*3)
}

File: src/fizzbuzz.swift
// FizzBuzzの値を返す関数 --- (*1)
func fizzbuzz(i:Int) -> String {
  if i % 3 == 0 && i % 5 == 0 { return "FizzBuzz" }
  if i % 3 == 0 { return "Fizz" }
  if i % 5 == 0 { return "Buzz" }
  return String(i)
}
// 1から100まで繰り返し関数を呼ぶ --- (*2)
for num in 1...100 {
  print(fizzbuzz(i:num)) // --- (*3)
}



